tla/consensus/ccfraft.tla

```diff
@@ -645,6 +645,10 @@ ChangeConfigurationInt(i, newConfiguration) ==
     \* i.e., be re-added to a new configuration.  Instead, the node has to rejoin with a
     \* "fresh" identity (compare sec 6.2, page 8, https://arxiv.org/abs/2310.11559).
     /\ \A s \in newConfiguration: s \notin removedFromConfiguration
+    \* See raft.h:2401, nodes are only sent future entries initially, they will NACK if necessary.
+    \* This is because they are expected to start from a fairly recent snapshot, not from scratch.
+    /\ LET addedNodes == newConfiguration \ CurrentConfiguration(i)
+       IN IF Cardinality(addedNodes) > 0 THEN \A addedNode \in addedNodes : sentIndex' = [sentIndex EXCEPT ![i][addedNode] = Len(log[i])] ELSE sentIndex' = sentIndex
```

Comment:
@lemmy  does `\A addedNode \in addedNodes : sentIndex' = [sentIndex EXCEPT ![i][addedNode] = Len(log[i])]` work alright?

Reply:
That's what I had before, but it sometimes resulted in sentIndex not being assigned (where there are no addedNodes, afaict), and TLC did treat that as an error.

Reply:
That's actually not quite what I meant, I wanted to check that a comparison to a primed variable within a for-all works like that.

I think the alternative is something like: 
```tla
LET
new_sent_index_i = [ k \in Servers |-> IF k \in AddedNodes THEN Len(log[i]) ELSE sentIndex[i][k]]
IN
sentIndex' = [sentIndex EXCEPT ![i] = new_sent_index_i]
```

Reply:
That does look simpler, I'll make the change.